File: medplum-java/medplum-model/src/main/java/com/fhir/SubstanceSpecification_Representation.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSubstanceSpecification_Representation.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SubstanceSpecification_Representation {
  Optional<Attachment> attachment();

  Optional<String> representation();

  Optional<String> id();

  Optional<CodeableConcept> type();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  static ImmutableSubstanceSpecification_Representation.Builder builder() {
    return ImmutableSubstanceSpecification_Representation.builder();
  }
}

File: medplum-typescript/src/model/fhir/SubstanceSpecification_Representation.ts
import { Attachment } from "./Attachment";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";

export interface SubstanceSpecification_Representation {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: CodeableConcept | null | undefined;
    representation: string | null | undefined;
    attachment: Attachment | null | undefined;
}
